### DISCUSSION TITLE
'nan' vs 'NaN' as default for CSV format

### DISCUSSION THREAD

**Discussion by youngsofun** [databendlabs/databend#8934] [General]
background:

https://github.com/datafuselabs/databend/issues/8913
https://github.com/datafuselabs/databend/issues/8914

**Comment by youngsofun** [comment]
we can limit it to only a few valid choices for now: "nan" or "NaN" or "null" or "NULL" ?

**Reply by youngsofun** [reply]
all so including "Nan"

**Comment by youngsofun** [comment]
https://note.nkmk.me/en/python-pandas-nan-none-na/

the value we use be accept by most csv lib in languages.

need to check this manually.

**Comment by youngsofun** [comment]
related https://github.com/datafuselabs/databend/discussions/8941

**Comment by youngsofun** [comment]
maybe we can choose to unload NaN by default and can be configed with option.
and accept all variants in load, no need to config.

**Reply by ariesdevil** [reply]
I think that's the appropriate way to deal with data.

**Reply by youngsofun** [reply]
there is a little problem with “NULL”  display
for column `float NULL`,  can not tell apart Nan from NULL

**Reply by youngsofun** [reply]
and clickhouse TSV may require `nan`,  we have to make clickhouse driver work.

need to check the code of clickhouse clients.
